compilation error on Windows with MSYS2 : undefined reference to `time_now_sec'

I compile nghttp2 1.57.0 this morning on Windows 10, with msys2, using the autotools.

config.h defines both `HAVE_CLOCK_GETTIME` and `HAVE_GETTICKCOUNT64`. Hence:

```
nghttp2_time.c:53:12: warning: implicit declaration of function 'time_now_sec' [-Wimplicit-function-declaration]
C:/Documents/msys2/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: .libs/nghttp2_time.o:nghttp2_time.c:(.text+0x29): undefined reference to `time_now_sec'
```

to solve this, I propose to change the order of the #ifdef in nghttp2_time.c. Something like

```
ifdef HAVE_GETTICKCOUNT64
uint64_t nghttp2_time_now_sec(void) { return GetTickCount64() / 1000; }
#elif defined(HAVE_CLOCK_GETTIME)
uint64_t nghttp2_time_now_sec(void) {
  struct timespec tp;
  int rv = clock_gettime(CLOCK_MONOTONIC, &tp);

  if (rv == -1) {
    return time_now_sec();
  }

  return (uint64_t)tp.tv_sec;
}
#else  /* !HAVE_CLOCK_GETTIME && !HAVE_GETTICKCOUNT64 */
uint64_t nghttp2_time_now_sec(void) { return time_now_sec(); }
#endif /* !HAVE_CLOCK_GETTIME && !HAVE_GETTICKCOUNT64 */
```

--- a/lib/nghttp2_time.c
+++ b/lib/nghttp2_time.c
@@ -44,7 +44,9 @@ static uint64_t time_now_sec(void) {
 }
 #endif /* HAVE_GETTICKCOUNT64 */
 
-#ifdef HAVE_CLOCK_GETTIME
+#ifdef HAVE_GETTICKCOUNT64
+uint64_t nghttp2_time_now_sec(void) { return GetTickCount64() / 1000; }
+#elif defined(HAVE_CLOCK_GETTIME)
 uint64_t nghttp2_time_now_sec(void) {
   struct timespec tp;
   int rv = clock_gettime(CLOCK_MONOTONIC, &tp);
@@ -55,8 +57,6 @@ uint64_t nghttp2_time_now_sec(void) {
 
   return (uint64_t)tp.tv_sec;
 }
-#elif defined(HAVE_GETTICKCOUNT64)
-uint64_t nghttp2_time_now_sec(void) { return GetTickCount64() / 1000; }
 #else  /* !HAVE_CLOCK_GETTIME && !HAVE_GETTICKCOUNT64 */
 uint64_t nghttp2_time_now_sec(void) { return time_now_sec(); }
 #endif /* !HAVE_CLOCK_GETTIME && !HAVE_GETTICKCOUNT64 */

